src/txn/txn.c

```diff
@@ -964,32 +953,30 @@ __txn_fixup_prepared_update(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, WT_U
      */
     F_SET(txn, WT_TXN_PREPARE_IGNORE_API_CHECK);
 
+    /*
+     * Set the stop time point to be the committing transaction's time point and copy the start time
+     * point from the current history store update.
+     */
     tw.stop_ts = txn->commit_timestamp;
     tw.durable_stop_ts = txn->durable_timestamp;
     tw.stop_txn = txn->id;
-    WT_TIME_WINDOW_SET_START(&tw, fix_upd);
+    tw.start_ts = hs_tw->start_ts;
```

Comment:
Use timestamp from the history store cell instead of using an update to pass the information.